sycl/test/kernel-and-program/kernel_cache.cpp

```diff
@@ -0,0 +1,148 @@
+// RUN: %clangxx -fsycl %s -o %t.out -lOpenCL
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+//==------------- program_cache.cpp - SYCL kernel/program test -------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <CL/sycl.hpp>
+
+class Functor {
+  public:
+  void operator()(cl::sycl::item<1> Item) {
+      (void)Item;
+  }
+};
+
+struct TestContext {
+  int Data;
+  cl::sycl::queue Queue;
+  cl::sycl::buffer<int, 1> Buf;
+
+  TestContext()
+    : Data(0), Buf(&Data, cl::sycl::range<1>(1))
+  {}
+
+  cl::sycl::program getProgram(const cl::sycl::string_class &BuildOptions = "") {
+    cl::sycl::program Prog(Queue.get_context());
+
+    Prog.build_with_kernel_type<class SingleTask>();
+
+    assert(Prog.get_state() == cl::sycl::program_state::linked);
```

Comment:
Please add meaningful message in asserts, like 
`assert(Prog.get_state() == cl::sycl::program_state::linked && "Linked state was expected");` or something like this.